presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java

```diff
@@ -93,6 +96,10 @@ private boolean isSupportedType(Type type)
             checkArgument(typeParameters.get(0) instanceof VarcharType, "Unsupported column type '%s' for map key", typeParameters.get(0));
             return true;
         }
+
+        if (type instanceof RowType) {
```

Comment:
I wonder if you modify avro decoder if you don't need to check or add tests for `redis` connector that is also using this.